app/controllers/work_package_children_controller.rb

```diff
@@ -49,9 +49,9 @@ def create
     target_work_package_id = params[:work_package][:id]
     target_child_work_package = WorkPackage.find(target_work_package_id)
 
-    target_child_work_package.parent = @work_package
-
-    if target_child_work_package.save
+    service_result = WorkPackages::UpdateService.new(user: current_user, model: target_child_work_package)
+                                                .call(parent: @work_package)
+    if service_result.success?
```

Comment:
The mistake that existed here before was repeated in the destroy action. In there, the service should be used as well. Otherwise a parent with two children with dates will keep the min/max of both children even if one is later on removed as a child.
So your hunch that there might be more amis was correct.

Reply:
Good catch.

Reply:
Fixed